netx/netx.go

```diff
@@ -308,6 +306,7 @@ func NewDNSClientWithOverrides(config Config, URL, hostOverride, SNIOverride str
 		c.Resolver = resolver.NewSerialResolver(txp)
 		return c, nil
 	case "dot":
+		config.TLSConfig = &tls.Config{NextProtos: []string{"dot"}}
```

Comment:
```suggestion
		config.TLSConfig.NextProtos = []string{"dot"}
```

We don't want to create a new config here. If we do, we loose the SNI override above.

Reply:
Thanks. done!